src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeVTableNode.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.TypeSystem;
+using System;
+using System.Collections.Generic;
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    internal class EETypeVTableNode : EmbeddedObjectNode
+    {
+        TypeDesc _type;
+        private bool ShouldBuildFullVTable { get; set; }
+        List<MethodDesc> _slots = new List<MethodDesc>();
+
+        public EETypeVTableNode(TypeDesc type)
+        {
+            _type = type;
+        }
+
+        protected override void OnMarked(NodeFactory context)
+        {
+            if (context.CompilationModuleGroup.ShouldProduceFullType(_type))
+            {
+                ShouldBuildFullVTable = true;
```

Comment:
I'm not sure I understand why we need to do this in OnMarked. Shouldn't the node already know if it will be on demand built or full when we construct it? And if so, we should be able to also get rid of BuildFullVTable and make it regular static dependencies.